docs/src/architecture/notes.md

```diff
@@ -1,73 +1,194 @@
-
-In `miden-lib` there are predefined P2ID and P2IDR note scripts that every user can simply create, see [code](https://github.com/0xPolygonMiden/miden-base/blob/fa63b26d845f910d12bd5744f34a6e55c08d5cde/miden-lib/src/notes/mod.rs#L15-L66). The script ensures that only the specified target account can consume the note and receive the asset. 
-
-The Note scripts is also the root of a [Miden program MAST](https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/main.html) which means every function is a commitment to the underlying code. The code cannot change unnoticed to the user because its hash would change.
-
-# Note recipient 
-We define `recipient` as: `hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash)` represented as `Word`. The account that executes a given note needs to provide the pre-image data to `recipient` during the transaction prologue. That means, one can create notes that can only be consumed if the `serial_num` and other data is known. This information can be passed on off-chain by the sender to the recipient. 
-
-# Note storage modes
-Similar to accounts, there are two storage modes for notes in Miden. Notes can be stored privately in the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database) with only the note hash. Or notes can be stored publicly with all data.
-
-Privately stored notes can only be consumed if the note data is known to the consumer. That means, there must be some offchain communication to transmit the note's data from the sender to the receipient.
-
-# Note metadata
-For every note the Miden Operator stores metadata in the Note DB. This metadata includes:
-
-* A **user-defined tag** as a means to quickly grab all notes for a certain application or use case.
-* A **sender** to be able to provide also ERC20 contract functionality.
-* The **number of assets** contained in the note. 
-
-# Note hash
-The note hash is computed as:
+ In addition, a note has **metadata** including the sender, tag, and number of assets.
 
-`hash(hash(hash(hash(serial_num, [0; 4]), script_hash), input_hash), vault_hash)`
+# Note's lifecycle
+New notes are being created when executing a transaction. After verifying the transaction proof the Operator adds either only the note hash or the full note data to the [Notes DB](https://0xpolygonmiden.github.io/miden-base/architecture/state.html#notes-database). Notes can be produced and consumed locally by users in local transactions or by the operator in a network transaction. Note consumption requires the transacting party to know the note data to compute the nullifier. After successful verification, the Operator sets the corresponding entry in the Nullifier DB to `1`. 
```

Comment:
Change to by executing transactions. 

Change Operator to "rollup" or "network". 

Explain why it would add only the hash or gull data. 

I would change local to private and network to public

Why `1`? We could explain that 1 = consumed.